cmake/SetupCharm.cmake

```diff
@@ -3,9 +3,13 @@
 
 # Apple Silicon (arm64) Macs require charm 7.0.0 to run
 if(APPLE AND "${CMAKE_HOST_SYSTEM_PROCESSOR}" STREQUAL "arm64")
-  find_package(Charm 7.0.0 EXACT REQUIRED)
+  find_package(Charm 7.0.0 REQUIRED)
 else()
-  find_package(Charm 6.10.2 EXACT REQUIRED)
+  find_package(Charm 6.10.2 REQUIRED)
+endif()
+if(CHARM_VERSION VERSION_GREATER 6.10.2)
+  message(STATUS "Builds with Charm++ versions greater than 6.10.2 are \
```

Comment:
I would prefer this be marked as `WARNING`. I know we do a lot less pre-processing of Charm++ generated files nowadays, but that's still subject to breaking and pretty fragile.